crypt4gh/__main__.py

```diff
@@ -11,36 +11,51 @@
 from .engine import encrypt, decrypt, reencrypt
 from .cli import parse_args
 from .keys import generate, get_public_key, get_private_key
+from .aws import get_parameter
 
 LOG = logging.getLogger(__name__)
 
 DEFAULT_PK  = os.getenv('C4GH_PUBLIC_KEY', None)
 DEFAULT_SK  = os.getenv('C4GH_SECRET_KEY', None)
+tmp_pk_file = "pk.pem"
 
 def run(args):
-
     # Parse CLI arguments
     args = parse_args(args)
 
     #####################################
-    ## For Encryption
+    ## Retrieve private key
     ##################################### 
-    if args['encrypt']:
-
-        seckey = args['--sk'] or DEFAULT_SK
-        if not seckey:
-            raise ValueError('Secret key not specified')
-        seckey = os.path.expanduser(seckey)
-        if not os.path.exists(seckey):
-            raise ValueError('Secret key not found')
-        cb = partial(getpass, prompt='Passphrase for {}: '.format(os.path.basename(args["--sk"])))
+    if args['encrypt'] or args['decrypt'] or args['reencrypt']:
+        if args['--kms_secret_id']:
+            with open("pk.pem", 'w') as f: 
+                f.write(get_parameter(args['--kms_secret_id']))
+            seckey = tmp_pk_file
+            cb = partial(getpass, prompt='Passphrase for {}: '.format(os.path.basename(args["--kms_secret_id"])))
+        else:
+            seckey = args['--sk'] or DEFAULT_SK
+            if not seckey:
+                raise ValueError('Secret key not specified')
+            seckey = os.path.expanduser(seckey)
+            if not os.path.exists(seckey):
+                raise ValueError('Secret key not found')
+            cb = partial(getpass, prompt='Passphrase for {}: '.format(os.path.basename(args["--sk"])))
+        
```

Comment:
Good idea to move the "fetch private key" code outside.
However, we can refactor it even further.
I can give it a try if you want, can I push changes to your branch?

Reply:
Sure thing. I've added you as a collaborator.

Reply:
I received the invitation, thanks. However, I realised that I can copy your branch and adjust here. Maybe that'll be simpler for me.

Reply:
👍

Reply:
I can invite you to that repo too, if you want. I'll add you as a collaborator.